### DISCUSSION TITLE
建议把需翻墙打开的网站自动加入自动规则，而不是需要手动加入

### DISCUSSION THREAD

**Discussion by Just2co** [SeaHOH/GotoX#26] [点子]
这个软件打开谷歌系网站飞快，但是打开其他需翻网站，非常慢，初步排除是DNS的问题，因为用其他GAE代理，解析都是飞快的，最后只能断定是这个软件默认会直连，连不上才会用GAE代理，而这个过程又是python程序来判断的，这个判断过程非常慢，通常要卡几十秒，听海大讲要把网站放进自动代理列表，GFWlist里的网站何其的多啊？一条条添加得到何时呢？

我现在找到了一个方案，就是把user.config.ini里的自动代理跟GAE代理的端口互换了一下，这样默认就是通过GAE代理打开了，排除了卡顿的问题，又不用逐条添加被墙网站，国内网站可以通过浏览器插件排除然后直连！

**Comment by Just2co** [comment]
@SeaHOH 直连转发是什么鬼？GAE代理我了解，就是这个软件还能通过GAE服务器直接转发网站数据？

**Comment by SeaHOH** [comment]
感谢 @Just2co 的建议，说下我为什么不默认加入一个的类似 gfwlist 的规则到 GAE 规则中：
* 首先 GotoX 是个超小众工具，我希望使用的人是看中了它的特色功能；现在只完成了一个自动代理，就是要自己折腾，通过它可以不必改代码也能解决某些特殊问题。
* 修改自动代理规则无需重启 GotoX，非常方便；一个人经常访问的网站也就那么几个，我不喜欢太多无关东西。
* 我已经设计了 15 分钟的临时规则来缓解此问题，还不至于无法访问，只是有几秒钟延迟，通常不超过 10 秒。

**Comment by SeaHOH** [comment]
直连转发就是直连转发，无关 GAE。

**Comment by Just2co** [comment]
@SeaHOH 能否给我讲解一下直连转发的原理？如果一个网站，在墙问无法访问，直连转发？是通过GAE来转发的么？那同GAE代理有啥区别？还是说在GAE上面搭建了一个隧道代理，直接类似socks一样的隧道？

**Comment by SeaHOH** [comment]
不经处理，**直连**转发。

**Comment by Just2co** [comment]
明白了， 也就是不经过代理，而是通过gotox客户端来转发数据，前提是能直连该网站，如果不能直连，还是得通过GAE代理来访问。

另外，这个15分钟规则在哪里改？想把时间调长一点，有时候访问一些娱乐网站时间比较长，谢谢。

**Comment by caofengzi** [comment]
這個東西有幾個優點，1是可以像goa那樣滿速下載，訪問網站的速度又超過goa，2它的速度可以和gop媲美還可以滿速下載，沒有像gop那樣下載大文件無法下載的問題。

**Comment by Just2co** [comment]
@caofengzi 访问网站的速度超过GOA,应该是IP筛选的功劳吧？另外，自动加入15分钟GAE代理，这个时间在哪里可以调整？

**Comment by SeaHOH** [comment]
@Just2co 
修改以下代码，不要使用 Windows 记事本。
```python
#ProxyHandler.py
    def go_GAE(self):
                #记录临时规则的过期时间
                filters_cache[host][-1] = '', '', 'TEMPGAE', time() + 900

    def go_FAKECERT(self):
                #设置临时规则的过期时间
                ssl_filters_cache.set(host, ('do_FAKECERT', None), 900)
```
